src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala

```diff
@@ -63,16 +59,23 @@ object SchemaSubject{
     recordNamespace: String
   ): SchemaSubject = {
     val dummySchema = createDummySchema(recordName, recordNamespace)
-    new SchemaSubject(TOPIC_RECORD_NAME_STRATEGY.subjectName(topicName, false, dummySchema))
+    new SchemaSubject(topicName + "-" + getRecordName(dummySchema))
   }
 
   def usingTopicRecordNameStrategy(
     topicName: String,
     schema: Schema
   ): SchemaSubject = {
-    new SchemaSubject(TOPIC_RECORD_NAME_STRATEGY.subjectName(topicName, false, schema))
+    new SchemaSubject(topicName + "-" + getRecordName(schema))
   }
 
+  private def getRecordName(schema: Schema): String =
+    if (schema.getType == Type.RECORD) {
+      schema.getFullName
+    } else {
+      throw new IllegalArgumentException("Schema must be of type RECORD")
```

Comment:
Please add the actual type to the error message

Reply:
done